firestore: marshal values as []byte, not string

awly opened the pull request:
Firestore marshaler requires all string data to be valid UTF-8. Our
values are sometimes binary (like QR codes for OTP signup), which would
fail to marhal.

On reads, try using both new and old formats to support existing data.

Added tests for this fallback and for writing binary data into all
backends.

Tested with firestore, including upgrade from an existing teleport cluster.

Fixes #4145

Commit:
firestore: marshal values as []byte, not string

Firestore marshaler requires all string data to be valid UTF-8. Our
values are sometimes binary (like QR codes for OTP signup), which would
fail to marhal.

On reads, try using both new and old formats to support existing data.

Added tests for this fallback and for writing binary data into all
backends.

webvictim commented:
Looks like the linter is failing.

awly commented:
Fixed linter failures

fspmarshall commented on lib/backend/firestore/firestorebk.go:
```diff
@@ -110,13 +110,63 @@ type FirestoreBackend struct {
 }
 
 type record struct {
+	Key       string `firestore:"key,omitempty"`
```
Should `Key` be changed to `[]byte` as well while we're here?  I don't think we enforce UTF-8 on keys either.

awly replied:
Good point. In practice it seems to always be UTF-8, but it's not enforced anywhere.
Changed and added test coverage for other backends.

Commit:
firestore: marshal keys as []byte

Same as values, `backend.Item` passes keys as `[]byte` and has no
guarantees about the encoding. GetRange queries need extra care because
of this type mismatch too: `where` clauses are type-sensitive.
